### PR TITLE
Document asset entities

### PR DESCRIPTION
**Pull request by agzam**
Closes: https://github.com/threatgrid/iroh/issues/4686

Document Asset, AssetProperties, AssetMapping, TargetRecord entities

## No QA required

### CODE REVIEW

**Review by ereteog — COMMENTED** [review]
You did not understand my explanations :-).
Let me try to rephrase it. When you add an `entry` to a flanders schema, you use the `entry` fn that takes the name of the field and its type as parameter and you can pass 2 other parameters, `:description` and a `string`, to document this entry.
Here is a concrete example from the [incident schema](https://github.com/threatgrid/ctim/blob/f9e2eff38e51ca0c2b8c4099457a1ecf07c8dba3/src/ctim/schemas/incident.cljc#L34):
```clojure
(f/entry :confidence v/HighMedLow
         :description (str "level of confidence held in the characterization "
                           "of this Incident"))
```
This task consists in adding this `description` parameters to document the fields of the asset entities. So for instance to take a very similar entry in `asset-mapping`:
``` clojure
(f/entry :confidence v/HighMedLow)
```
you can change it to:
``` clojure
(f/entry :confidence v/HighMedLow
             :description (str "level of confidence held in the characterization "
                               "of this AssetMapping"))
```
And then you continue with all other field. Then you can regenerate the doc with `lein doc` like you did to have well docuemented fields for asset entities.

File: project.clj [hunk 1]
```diff
@@ -6,7 +6,7 @@
   :url "http://github.com/threatgrid/ctim"
   :license {:name "Eclipse Public License"
             :url "http://www.eclipse.org/legal/epl-v10.html"}
-  :pedantic? :abort
+  ;; :pedantic? :abort
```
Comment (ereteog) [inline project.clj:9]: could you uncomment it please?

File: src/ctim/schemas/asset_mapping.cljc [hunk 2]
```diff
@@ -26,13 +27,26 @@
...
-   (f/entry :stability Stability)
-   (f/entry :observable c/Observable)
-   (f/entry :asset_ref asset/AssetRef)
-   (f/entry :asset_type asset/AssetType)))
+   (f/entry :valid_time c/ValidTime
+            :description (str "For each asset, we allow for the assertion of time bound properties."
+                              "This gives us both a record of the current state of the asset,"
+                              "as well as history."))
+   (f/entry :confidence v/HighMedLow
+            :description "Is it susceptible to manipulation or translation.")
```
Comment (ereteog) [inline src/ctim/schemas/asset_mapping.cljc:35]: I would be more general like in [Incident](https://github.com/threatgrid/ctim/blob/f9e2eff38e51ca0c2b8c4099457a1ecf07c8dba3/src/ctim/schemas/incident.cljc#L35) and you could use your description as an example):
"level of confidence held in the characterization of this AssetMapping (e.g. is it susceptible to manipulation or translation?)."

File: src/ctim/schemas/asset_mapping.cljc [hunk 3]
```diff
@@ -26,13 +27,26 @@
...
+   (f/entry :specificity Specificity
+            :description (str "Denotes the level of how many assets "
+                              "potentially could have this same identifier."))
+   (f/entry :stability Stability
+            :description (str "Do we manage when it changes, "
+                              "or it always a time bound assignment?"))
```
Comment (ereteog) [inline src/ctim/schemas/asset_mapping.cljc:41]:
```suggestion
                              "or is it always a time bound assignment?"))
```

File: src/ctim/schemas/asset_properties.cljc [hunk 4]
```diff
@@ -25,8 +32,10 @@
   c/sourced-object-entries
   (f/required-entries
    (f/entry :type AssetPropertiesTypeIdentifier)
-   (f/entry :valid_time c/ValidTime)
-   (f/entry :asset_ref asset/AssetRef))
+   (f/entry :valid_time c/ValidTime
+            :description "The time range during which the AssetProperties considered valid.")
```
Comment (ereteog) [inline src/ctim/schemas/asset_properties.cljc:36]:
```suggestion
            :description "The time range during which the AssetProperties is considered valid.")
```

File: src/ctim/schemas/target_record.cljc [hunk 5]
```diff
@@ -18,9 +18,14 @@
   "Schema for TargetRecord Targets"
   (:entries c/IdentitySpecification)
   (f/optional-entries
-   (f/entry :os f/any-str)
-   (f/entry :internal (f/bool :default false))
-   (f/entry :sensor f/any-str)
+   (f/entry :os f/any-str
+            :description (str "Source Operating System where TargetRecord was originated."))
+   (f/entry :internal (f/bool :default false)
+            :description "Is it internal to our network.")
```
Comment (ereteog) [inline src/ctim/schemas/target_record.cljc:24]:
```suggestion
            :description "Is it internal to our network?")
```

**Review by ereteog — APPROVED** [review]
LGTM

File: project.clj (hunk 1 above)
Comment (agzam) [reply project.clj:9]: Ah shoot... Good catch... Sorry about that.

**Pull request merged**